hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOzoneManagerHAWithData.java

```diff
@@ -109,6 +109,7 @@ public void testFileOperationsAndDelete() throws Exception {
     testFileOperationsWithRecursive();
     testFileOperationsWithNonRecursive();
     testKeysDelete();
+    Thread.sleep(1000);
```

Comment:
Can you please explain why sleep is needed?  What do we need to wait for?  Would it be possible to explicitly wait for that condition?

https://github.com/apache/ozone/blob/5ed0e0a9b4c8355f4cde064cc605bc419dacd9c3/hadoop-hdds/test-utils/src/main/java/org/apache/ozone/test/GenericTestUtils.java#L192-L228

Reply:
@symious, another util you can use to check a condition between intervals until a timeout is the `LamdaTestUtils.await`:
ie.  `LambdaTestUtils.await(WAIT_TIMEOUT_MILLIS, 1000, () -> {...}`

Reply:
@adoroszlai @neils-dev Thanks for the review.
The `Thread.sleep` is a walkaround to keep the ratis follower in a stable state. But the root cause should be the restarting of RPC server introduced in HDDS-6685.

Reply:
Thanks @symious for the reply, but this still does not explain why explicit wait for some condition is not possible instead of sleep.

Reply:
I tried to solve the timeout issue caused by HDDS-6685. But the error is still there, only the error message is not "port already in use", but "java.lang.IllegalStateException: gap between start index 131 and first entry to append 218".

I think it's incurred by ratis-2.3.0, seems in ratis-2.2.0, the error of loading of SNAPSHOT is handled differently, so with ratis-2.2.0, the error was solved but with ratis-2.3.0 it's incurring some new errors.

Reply:
I think "the sleep" is not a good way to solve this flasky test.